db/commitlog/commitlog.cc

```diff
@@ -653,15 +653,26 @@ class db::commitlog::segment : public enable_shared_from_this<segment>, public c
          * queue, just to be sure.
          */
         auto me = shared_from_this();
-        return close().finally([me] {
-            // When we get here, nothing should add ops,
-            // and we should have waited out all pending.
-            return me->_pending_ops.close().finally([me] {
-                return me->_file.truncate(me->_flush_pos).then([me] {
-                    return me->_file.close().finally([me] { me->_closed_file = true; });
-                });
-            });
-        });
+        // could have kept the "finally" continuations
+        // here, but this potentially missed immediate 
+        // exceptions thrown in close/p_o.close.
+        std::exception_ptr p;
+        try {
+            co_await close();
+        } catch (...) {
+            p = std::current_exception();
+        }
+    
+        co_await _pending_ops.close();
+        co_await _file.truncate(_flush_pos);
+        co_await _file.close();
```

Comment:
This exposes a pre-existing problem that if truncate fails, then we won't close(). This is a small problem since truncate is unlikely to fail, and if it does fail, we'll shut down the node and the unclosed file will be a small problem. Still, it's a minor bug that is better fixed (later).

Reply:
Yes, I opted to keep behaviour identical, issues and all.